src/Orchard/Tasks/Locking/Services/DistributedLockService.cs

```diff
@@ -141,9 +141,19 @@ public class DistributedLockService : Component, IDistributedLockService {
 
             ExecuteOnSeparateTransaction(repository => {
                 // Try to find a valid lock record in the database.
-                var record = repository.Table.FirstOrDefault(x => x.Name == internalName && (x.ValidUntilUtc == null || x.ValidUntilUtc >= _clock.UtcNow));
+                var records = repository.Table.Where(x => x.Name == internalName).ToList();
+                var record = records.FirstOrDefault(x => x.ValidUntilUtc == null || x.ValidUntilUtc >= _clock.UtcNow);
                 if (record == null) {
-                    // No record existed, so we're good to create a new one.
+                    if (records.Any()) {
```

Comment:
It is not. One is record without a "s" and the other is records with a "s"
It validates if "record" has not found any results. Then it looks in the "records" list to delete it.
I suggest renaming the vars to avoid eventual confusion ;)